public/assets/ts/admin/toast.ts

```diff
@@ -0,0 +1,19 @@
+// the disable will be removed once we can import/export modules in the frontend
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function showToast(message: string, intent: "success" | "fail") {
+  const toast = document.getElementById("toast");
+
+  const toastHeader = document.querySelector(".toast-header");
+  const toastMessage = document.querySelector(".toast-message");
+  const toastBody = document.querySelector(".toast-body");
+
+  toastHeader.innerHTML = intent === "success" ? "Success" : "Failure";
+  toastMessage.innerHTML = message;
+  toast.classList.add("visible");
+  toastBody.classList.add(intent);
+
+  setTimeout(() => {
+    toast.classList.remove("visible");
```

Comment:
There's some weird behaviour that happens when someone tries to click multiple times while a toast is already onscreen, where the toast doesn't stay up for the full amount of time.  

Would it be possible to update this function to store the most recent click time, then only close if 2000ms have since passed?  That is, if two timeouts are going simultaneously, we make it so that the timeout that finishes first doesn't actually close the modal because it knows the second timeout is still running.

Reply:
i'm not really a fan of this solution since this toast may not always be shown after something is clicked, could be shown for any reason - but I came up with another sol to stop the weird behvaior.